### PR TITLE
Implemented fill for raster images.

### PR DESCRIPTION
**Pull request by davidpiuva**
It should now be possible to fill raster images with colors.

Since filling a YUV image is usually to the whole image and not individual components, filling monochrome or UV images only takes the first channels from the RGBA color just like the GPU does.

Since RGB cannot be loaded back to the CPU, I could not test the already implemented RGB fill on the GPU.

Solves https://github.com/magic-lang/ooc-kean/issues/1411

### CODE REVIEW

**Comment by davidpiuva** [comment]
@emilwestergren Peer review?

File: source/draw/RasterMonochrome.ooc [hunk 1]
```diff
@@ -40,6 +40,11 @@ RasterMonochromeCanvas: class extends RasterPackedCanvas {
 		if (monochrome != image)
 			monochrome referenceCount decrease()
 	}
+	fill: override func (color: ColorRgba) {
```
Comment (emilwestergren) [inline source/draw/RasterMonochrome.ooc]: Should be optimized with a single `memset` later.

File: source/draw/RasterUv.ooc [hunk 2]
```diff
@@ -40,6 +40,11 @@ RasterUvCanvas: class extends RasterPackedCanvas {
 		if (uv != image)
 			uv referenceCount decrease()
 	}
+	fill: override func (color: ColorRgba) {
```
Comment (emilwestergren) [inline source/draw/RasterUv.ooc]: This can be optimized with `memset` as well, if `u == v`

**Comment by davidpiuva** [comment]
Only a time-out error in the promise test that is not related to raster fill.

File: source/draw/RasterMonochrome.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/draw/RasterMonochrome.ooc]: Why not now? It's an easy fix and it's one line of code

Comment (davidpiuva) [inline source/draw/RasterMonochrome.ooc]: Because reverting is easier if we have something to fall back on.

File: test/draw/gpu/FillTest.ooc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+use collections
+use geometry
+use draw-gpu
+use draw
+use opengl
+use unit
+
+FillTest: class extends Fixture {
+	init: func {
+		super("FillTest")
```
Comment (marcusnaslund) [inline test/draw/gpu/FillTest.ooc]: Just `"Fill"`

File: source/draw/RasterMonochrome.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/draw/RasterMonochrome.ooc]: But we're not even using this code at the moment.

Comment (davidpiuva) [inline source/draw/RasterMonochrome.ooc]: I can make a separate commit then.

Comment (marcusnaslund) [inline source/draw/RasterMonochrome.ooc]: That's fine.

File: source/draw/RasterMonochrome.ooc [hunk 4]
```diff
@@ -40,6 +40,9 @@ RasterMonochromeCanvas: class extends RasterPackedCanvas {
...
+		this target buffer memset(color r)
```
Comment (davidpiuva) [inline source/draw/RasterMonochrome.ooc]: Updated

**Comment by davidpiuva** [comment]
@marcusnaslund All checks passed.

File: source/base/ByteBuffer.ooc [hunk 5]
```diff
@@ -32,8 +32,10 @@ ByteBuffer: class {
 		this _pointer = null
 		super()
 	}
-	zero: func ~whole { memset(this _pointer, 0, this _size) }
-	zero: func ~range (offset, length: Int) { memset(this _pointer + offset, 0, length) }
+	zero: func ~whole { this memset(0) }
+	zero: func ~range (offset, length: Int) { this memset(offset, length, 0) }
+	memset: func ~whole (value: Byte) { memset(this _pointer, value, this _size) }
+	memset: func ~range (offset, length: Int, value: Byte) { memset(this _pointer + offset, value, length) }
```
Comment (marcusnaslund) [inline source/base/ByteBuffer.ooc]: In light of the TDD mentality we're trying to enforce, I should ask you do add some small test in `ByteBufferTest` using `memset`. The rest of the PR looks good.

Comment (davidpiuva) [inline source/base/ByteBuffer.ooc]: Updated with two new tests.

**Comment by marcusnaslund** [comment]
Perfection.

**Comment by davidpiuva** [comment]
All checks passed again.

File: source/base/ByteBuffer.ooc (hunk 5 above)
Comment (marcusnaslund) [inline source/base/ByteBuffer.ooc]: I'm also not sure if these methods should be called `memset`, but we can discuss that another time.

**Comment by marcusnaslund** [comment]
Apparently, the keyword "Fixes", but not "Solves", automatically closes issues.

**Pull request merged**